pkg/webhook/router.go

```diff
@@ -54,8 +54,9 @@ type handlerRouter struct {
 }
 
 func (r *handlerRouter) Handle(ctx context.Context, req admission.Request) admission.Response {
-	if !utils.UserGroupList(req.UserInfo.Groups).IsInCapsuleGroup() {
-		// not a Capsule user, can be skipped
+
+	if !utils.UserGroupList(req.UserInfo.Groups).IsInCapsuleGroup() && req.Kind.Kind != "Tenant" {
```

Comment:
Because we need `tenant_name` webhook to always fire. Tenants can be created by admins, who can be not a members of `capsule-user-group`. Without this condition they will be able to create tenants, whose name is not validated by `tenant_name` webhook

Reply:
Oh, I see... it seems we need a sort of Middleware here.
Sorry to block your PR here but we need to figure out a better webhook middleware piping.

Reply:
No problem :) For example, we can pass here
https://github.com/clastix/capsule/blob/9d845b367852fb19d1e2756fe7e27fd6338dcdf1/pkg/webhook/router.go#L51-L56
and additional map of `enforcedResourceKinds` with the Kinds of resources we want hooks to always fire for
e.g
```
enforcedResourceKinds := map[string]bool {
    "Tenant": true,
}
```
and here https://github.com/clastix/capsule/blob/9d845b367852fb19d1e2756fe7e27fd6338dcdf1/main.go#L141

use 
```
err = webhook.Register(mgr, capsuleGroup,enforcedResourceKinds, wl...)
```
so the code will be like 
```

if !utils.UserGroupList(req.UserInfo.Groups).IsInCapsuleGroup(r.capsuleGroup) && !enforcedResourceKinds[req.Kind.Kind]  {
		// not a Capsule user, not a tenant kind, can be skipped
		return admission.Allowed("")
	}
```

Reply:
I was thinking more about closures, here, as described [here](https://www.calhoun.io/pitfalls-of-context-values-and-how-to-avoid-or-mitigate-them/) although regarding the `context` API